### ISSUE TITLE
Last time Telegram Desktop was not properly closed

### ISSUE THREAD

**Issue by kitscribe** [telegramdesktop/tdesktop#27736]
### Steps to reproduce

1. Open an app
2. Switch chats a couple times
3. Right-click on a message to choose "Reply to"
4. Get a freeze and crashed app
5. Open an app again, see the "Checking for updates" window

### Expected behaviour

No crashes when trying to reply a message

### Actual behaviour

Get crashes when trying to reply a message or set a reaction

### Operating system

Debian 12 + KDE

### Version of Telegram Desktop

4.16.6

### Installation source

Static binary from official website

### Crash ID

be3f8f32-2114-4e6b-1c404e8a-c314a7fc

### Logs

```shell
Can not reproduce a bug for the report, but it seems like memory leak
```

**Comment by ilya-fedin** [comment]
> ### Crash ID
> 
> _No response_

This is required

**Comment by ilya-fedin** [comment]
> That is not required field within the form...

GitHub doesn't let to detect whether you report a crash and automatically make this field required... But it is required when you're reporting a crash.

**Comment by kitscribe** [comment]
> > That is not required field within the form...
> 
> GitHub doesn't let to detect whether you report a crash and automatically make this field required... But it is required when you're reporting a crash.

Well, okay, how can I make a report if the application does not offer to send it?

**Comment by ilya-fedin** [comment]
> Well, okay, how can I make a report if the application does not offer to send it?

Well, you can't... You're supposed to have a crash ID for a crash. If you don't have it, prehaps you haven't done the steps specified in the crash ID field description when creating the issue...

**Comment by kitscribe** [comment]
> prehaps you haven't done the steps specified in the crash ID field

The general flow that occurs can be described as follows:

The user opens the application, do his business, and suddenly, when he wanted to reply to a message or set a reaction, the application froze and closed after a while.

Then, when the user open the application again, it says that last time it was not closed correctly and displays a 'Checking for updates' window which is also comes with errors. Clicking 'TRY AGAIN' has no effect, so you have to click 'SKIP' to start the application. And when you click 'SKIP' to launch the app, it doesn't prompt you to submit a crash report.

So how can I create such a report in some other way?

**Comment by ilya-fedin** [comment]
> So how can I create such a report in some other way?

What's the problem to do what is explained when creating the issue?

**Comment by kitscribe** [comment]
> What's the problem to do what is explained when creating the issue?

Okay, I'm turning on a beta versions and wait for the crash again

**Comment by kitscribe** [comment]
> > So how can I create such a report in some other way?
> 
> What's the problem to do what is explained when creating the issue?

Updated the top message with crash report id

**Comment by kitscribe** [comment]
That is not a memory leak, but an incorrect accessing to it

**Comment by kitscribe** [comment]
Here is another one crash id: 853f3a0f-933f-4bd0-b4fbdcab-5f9edd60

**Comment by ilya-fedin** [comment]
Do you have NVIDIA?

**Comment by ilya-fedin** [comment]
The traces are very weird. The one in the issue itself:
```
Operating system: Linux
                  0.0.0 Linux 6.1.0-18-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.76-1 (2024-02-01) x86_64
CPU: amd64
     family 6 model 154 stepping 3
     1 CPU

GPU: UNKNOWN

Crash reason:  SIGSEGV /SEGV_MAPERR
Crash address: 0x7efbdb14df80
Process uptime: not available

Thread 0 (crashed)
 0  libzstd.so.1 + 0x8b02a
    rax = 0x00007efbdb14dfa0   rdx = 0x0000000000000000
    rcx = 0x00007efc3ae1adac   rbx = 0x00007efbdb14df80
    rsi = 0x0000000000000000   rdi = 0x00007efbdb14df80
    rbp = 0x0000000000000000   rsp = 0x00007ffd4ec9b930
     r8 = 0x000000000000008c    r9 = 0x0000000000000000
    r10 = 0x00007efc3ae1adac   r11 = 0x00007efc45a1d801
    r12 = 0x0000000000000000   r13 = 0x0000000000000348
    r14 = 0x000000000000008c   r15 = 0x0000000000000000
    rip = 0x00007efc413cd02a
    Found by: given as instruction pointer in context
 1  libzstd.so.1 + 0x8b092
    rsp = 0x00007ffd4ec9b940   rip = 0x00007efc413cd092
    Found by: stack scanning
 2  libzstd.so.1 + 0x8b5c5
    rsp = 0x00007ffd4ec9b960   rip = 0x00007efc413cd5c5
    Found by: stack scanning
 3  libzstd.so.1 + 0x8bb30
    rsp = 0x00007ffd4ec9ba10   rip = 0x00007efc413cdb30
    Found by: stack scanning
 4  libzstd.so.1 + 0x8bc51
    rsp = 0x00007ffd4ec9ba30   rip = 0x00007efc413cdc51
    Found by: stack scanning
 5  libgallium_dri.so + 0xc2c3b
    rsp = 0x00007ffd4ec9ba80   rip = 0x00007efc2c2c2c3b
    Found by: stack scanning
 6  libgallium_dri.so + 0xb6ba9
    rsp = 0x00007ffd4ec9ba90   rip = 0x00007efc2c2b6ba9
    Found by: stack scanning
 7  libgallium_dri.so + 0xb72fd
    rsp = 0x00007ffd4ec9bb10   rip = 0x00007efc2c2b72fd
    Found by: stack scanning
 8  libgallium_dri.so + 0xd94d64
    rsp = 0x00007ffd4ec9bc00   rip = 0x00007efc2cf94d64
    Found by: stack scanning
 9  libgallium_dri.so + 0xba877
    rsp = 0x00007ffd4ec9bc20   rip = 0x00007efc2c2ba877
    Found by: stack scanning
10  libgallium_dri.so + 0xd9f9e5
    rsp = 0x00007ffd4ec9bcc0   rip = 0x00007efc2cf9f9e5
    Found by: stack scanning
11  libgallium_dri.so + 0xda0257
    rsp = 0x00007ffd4ec9bd70   rip = 0x00007efc2cfa0257
    Found by: stack scanning
12  libgallium_dri.so + 0xda0257
    rsp = 0x00007ffd4ec9bd80   rip = 0x00007efc2cfa0257
    Found by: stack scanning
13  libgallium_dri.so + 0x182ddd
    rsp = 0x00007ffd4ec9be30   rip = 0x00007efc2c382ddd
    Found by: stack scanning
14  libgallium_dri.so + 0x18367d
    rsp = 0x00007ffd4ec9beb0   rip = 0x00007efc2c38367d
    Found by: stack scanning
```
Whch says it crashes in libzstd called by mesa. The trace also should go up to main() entrypoint yet it says libgallium_dri.so is the first frame.

The one in the comment is:
```
Crash reason:  SIGSEGV /SEGV_MAPERR
Crash address: 0x7f3eb7b0f040
Process uptime: not available

Thread 0 (crashed)
 0  libc.so.6 + 0x152a8a
    rax = 0x00007f3eb7b0f040   rdx = 0x00000000000beb58
    rcx = 0x00000000000beb58   rbx = 0x00000000000beb58
    rsi = 0x00007f3f1061c000   rdi = 0x00007f3eb7b0f040
    rbp = 0x00007ffc056ef930   rsp = 0x00007ffc056ef898
     r8 = 0x00007f3eb7b0f040    r9 = 0x0000000000000010
    r10 = 0x0000000000000004   r11 = 0x0000000000000002
    r12 = 0x00007f3ed3133918   r13 = 0x00007ffc056ef950
    r14 = 0x00007f3f1061c000   r15 = 0x00007f3ed34fa3d0
    rip = 0x00007f3f3be54a8a
    Found by: given as instruction pointer in context
 1  Telegram!QImage::detach() [qimage.cpp : 1115 + 0x5]
    rbp = 0x00007ffc056ef9a0   rsp = 0x00007ffc056ef940
    rip = 0x000055f0053f49d2
    Found by: previous frame's frame pointer
 2  Telegram!QPainter::begin(QPaintDevice*) [qpainter.cpp : 1714 + 0x8]
    rbx = 0x00007f3eec317fc0   rbp = 0x00007ffc056efa30
    rsp = 0x00007ffc056ef9b0   r12 = 0x00007f3ed3133918
    rip = 0x000055f0054e7db8
    Found by: call frame info
 3  Telegram!QtWaylandClient::QWaylandShmBackingStore::beginPaint(QRegion const&) [qwaylandshmbackingstore.cpp : 190 + 0x5]
    rbx = 0x00007f3ed3133900   rbp = 0x00007ffc056efaa0
    rsp = 0x00007ffc056efa40   r12 = 0x00007ffc056efa50
    r13 = 0x00007ffc056efa40   r14 = 0x00007ffc056efae8
    r15 = 0x00007f3ed34fa3d0   rip = 0x000055f0048a5f91
    Found by: call frame info
 4  Telegram!QBackingStore::beginPaint(QRegion const&) [qbackingstore.cpp : 125 + 0xe]
    rbx = 0x00007f3f2e56daa0   rbp = 0x00007ffc056efb50
    rsp = 0x00007ffc056efab0   r12 = 0x00007f3ed34fa3d0
    r13 = 0x00007ffc056efbd8   r14 = 0x00007ffc056efae8
    r15 = 0x000055f0048a5d60   rip = 0x000055f005409ea1
    Found by: call frame info
 5  Telegram!QWidgetRepaintManager::paintAndFlush() [qwidgetrepaintmanager.cpp : 869 + 0x5]
    rbx = 0x0000000000000000   rbp = 0x00007ffc056efe10
    rsp = 0x00007ffc056efb60   r12 = 0x0000000000000000
    r13 = 0x00007f3f3142b8a0   r14 = 0x00007ffc056efc00
    r15 = 0x00007f3ed30a09c0   rip = 0x000055f00500408f
    Found by: call frame info
 6  Telegram!QWidget::event(QEvent*) [qwidget.cpp : 1762 + 0x5]
    rbx = 0x00007f3effe8b580   rbp = 0x00007ffc056eff00
    rsp = 0x00007ffc056efe20   r12 = 0x00007f3effe8b800
    r13 = 0x00007f3eb6f07940   r14 = 0x00007ffc056eff10
    r15 = 0x00007f3ef248f300   rip = 0x000055f0050054d1
    Found by: call frame info
 7  Telegram!Ui::RpWidgetWrap::handleEvent(QEvent*) [rp_widget.cpp : 247 + 0x9]
    rbx = 0x00007f3effe8b5a8   rbp = 0x00007ffc056eff70
    rsp = 0x00007ffc056eff10   r12 = 0x00007f3ef248f300
    r13 = 0x00007f3eb6f07940   r14 = 0x00007ffc056eff10
    r15 = 0x000000000000004d   rip = 0x000055f003321cb0
    Found by: call frame info
```
This one is cropped only to the relevant part which says Qt tries to detach (copy) the window buffer while starting filling it with transparency which shouldn't happen in first place. It seem to crash on `*this = copy();` inside QImage::detach(). Again, I don't understand how this happens. This is deep inside Qt, going through the QPA (Qt Platform Abstraction).

**Comment by kitscribe** [comment]
> This is deep inside Qt, going through the QPA (Qt Platform Abstraction)

So, this is not a bug, but a feature or something? :)

If to be serious, it turns out that this is a bug in QT framework, but not in the client?
Is there anything I can do now or should I just wait for an update?

**Comment by ilya-fedin** [comment]
> If to be serious, it turns out that this is a bug in QT framework, but not in the client?

Not sure... Maybe it's some bug in OpenGL driver? Have you tried to disable OpenGL in Telegram settings?

**Comment by kitscribe** [comment]
> > If to be serious, it turns out that this is a bug in QT framework, but not in the client?
> 
> Not sure... Maybe it's some bug in OpenGL driver? Have you tried to disable OpenGL in Telegram settings?

Never, but I'll try...

**Comment by ilya-fedin** [comment]
Have it helped?

**Comment by kitscribe** [comment]
> Have it helped?

Since I had no problems, but I also haven't used desktop client since Thursday so I'm not sure if it a proper solution

**Comment by kitscribe** [comment]
> Have it helped?

Well, it seems like since there were no problems with that. Should I close the issue and mark it as solved?

**Comment by ilya-fedin** [comment]
Yeah... Apparently your driver is bugged :(

**Comment by kitscribe** [comment]
Disabling OpenGL helped to fix crashes

**Comment by ilya-fedin** [comment]
I managed to reproduce this. Looks like it started to happen since 10272ee0cf266f5d6c25259e4ac2e8341c952688. Not sure what to do about that though as this commit fixes #27446...
